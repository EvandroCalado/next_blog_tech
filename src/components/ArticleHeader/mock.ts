export default {
  id: '3',
  title: 'Desenvolvimento de aplicativos web escaláveis com Next.js',
  excerpt:
    'Se você está procurando um framework poderoso para desenvolver aplicativos web escaláveis, o Next.js pode ser a escolha certa para você. O Next.js é um framework baseado em React que oferece recursos avançados, como renderização no lado do servidor e roteamento dinâmico, para ajudá-lo a criar aplicativos web modernos e de alto desempenho. Neste post, exploraremos as principais características do Next.js e veremos como usá-lo para construir aplicativos web escaláveis.',
  createdAt: '2023-03-27T17:22:39.749Z',
  cover:
    'https://res.cloudinary.com/dnhsdenji/image/upload/v1679929576/twitter_card_f0df8b6a20.png',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
