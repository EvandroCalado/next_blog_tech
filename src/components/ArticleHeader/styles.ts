import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.small} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.small} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
