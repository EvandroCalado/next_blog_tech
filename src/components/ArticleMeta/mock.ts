export default {
  createdAt: '2023-03-27T17:22:39.749Z',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
