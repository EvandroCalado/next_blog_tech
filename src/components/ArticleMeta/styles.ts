import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child)::after {
      content: ', ';
    }

    span:first-child {
      margin-right: 0.5rem;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
