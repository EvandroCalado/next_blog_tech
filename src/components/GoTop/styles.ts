import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    position: fixed;
    bottom: 3rem;
    right: 5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    text-decoration: none;
    transition: all 300ms ease-in-out;

    &:hover {
      color: ${theme.colors.secondary};
    }
  `}
`;
