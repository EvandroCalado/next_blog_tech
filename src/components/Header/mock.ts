import { HeaderProps } from '.';

export default {
  blogName: 'Evandro Calado',
  blogDescription:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Unde harum optio, eius quia quas molestias ipsum modi sed ullam exercitationem.',
  logo: 'https://res.cloudinary.com/dnhsdenji/image/upload/v1679924887/110628201_ae0753864b.jpg',
  showText: true,
} as HeaderProps;
