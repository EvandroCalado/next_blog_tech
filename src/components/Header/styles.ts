import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.dark};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${Title} {
      margin: ${theme.spacings.small} 0;
      text-align: left;
    }

    @media ${theme.media.medium} {
      flex-flow: column wrap;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;

    @media ${theme.media.medium} {
      margin: 0;
    }
  `}
`;
