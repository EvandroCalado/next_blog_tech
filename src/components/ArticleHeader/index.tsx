import { Author } from '../shared-types/author';
import Heading from '../Heading';
import * as Styled from './styles';
import ArticleMeta, { Category } from '../ArticleMeta';

export type ArticleHeaderProps = {
  id: string;
  title: string;
  slug: string;
  excerpt: string;
  cover: string;
  author: Author;
  categories: Category[];
  createdAt: string;
};
const ArticleHeader = ({
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt,
}: ArticleHeaderProps) => {
  return (
    <Styled.Wrapper>
      <Heading as="h1" size="huge">
        {title}
      </Heading>
      <Styled.Excerpt>{excerpt}</Styled.Excerpt>
      <Styled.Cover src={cover} alt={title} />
      <ArticleMeta
        categories={categories}
        author={author}
        createdAt={createdAt}
      />
    </Styled.Wrapper>
  );
};

export default ArticleHeader;
