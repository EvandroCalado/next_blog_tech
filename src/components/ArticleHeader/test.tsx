import ArticleHeader from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';
import { formatDate } from '../utils/format-date';

describe('<ArticleHeader />', () => {
  it('should render heading, excerpt, cover, img and meta', () => {
    const { container } = renderTheme(<ArticleHeader {...mock} />);
    expect(
      screen.getByRole('heading', { name: mock.title }),
    ).toBeInTheDocument();
    expect(screen.getByRole('img', { name: mock.title })).toBeInTheDocument();
    expect(screen.getByText(mock.excerpt)).toBeInTheDocument();
    expect(screen.getByText(formatDate(mock.createdAt))).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
