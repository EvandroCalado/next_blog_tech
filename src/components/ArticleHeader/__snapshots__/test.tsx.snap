// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<ArticleHeader /> should render heading, excerpt, cover, img and meta 1`
] = `
.c2 {
  color: #2a3840;
  line-height: 1.2;
  font-size: 6.4rem;
  font-weight: 900;
  text-transform: none;
  text-align: center;
}

.c0 {
  padding-bottom: 4rem;
  margin-bottom: 4rem;
  border-bottom: 0.1rem solid #ddd;
}

.c0 .c1 {
  margin: 0;
  margin-bottom: 2.4rem;
}

.c3 {
  margin: 2.4rem 0;
  font-size: 2.4rem;
}

.c4 {
  width: 100%;
  display: block;
  margin-bottom: 2.4rem;
}

.c5 {
  font-size: 1.6rem;
  color: #aaa;
  font-style: italic;
}

.c5 .categories span:not(:last-child)::after {
  content: ', ';
}

.c5 span:first-child {
  margin-right: 0.5rem;
}

.c5 a {
  color: #f24b4b;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c5 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

@media (max-width:768px) {
  .c2 {
    font-size: 4rem;
  }
}

@media (max-width:360px) {
  .c0 .c1 {
    font-size: 2.4rem;
  }
}

@media (max-width:360px) {
  .c3 {
    font-size: 1.6rem;
  }
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      Desenvolvimento de aplicativos web escaláveis com Next.js
    </h1>
    <p
      class="c3"
    >
      Se você está procurando um framework poderoso para desenvolver aplicativos web escaláveis, o Next.js pode ser a escolha certa para você. O Next.js é um framework baseado em React que oferece recursos avançados, como renderização no lado do servidor e roteamento dinâmico, para ajudá-lo a criar aplicativos web modernos e de alto desempenho. Neste post, exploraremos as principais características do Next.js e veremos como usá-lo para construir aplicativos web escaláveis.
    </p>
    <img
      alt="Desenvolvimento de aplicativos web escaláveis com Next.js"
      class="c4"
      src="https://res.cloudinary.com/dnhsdenji/image/upload/v1679929576/twitter_card_f0df8b6a20.png"
    />
    <div
      class="c5"
    >
      <p>
        <span>
          Por
        </span>
        <a
          href="/author/evandro-calado"
        >
          Evandro Calado
        </a>
        <span
          class="separator"
        >
           | 
        </span>
        <time
          datetime="2023-03-27T17:22:39.749Z"
        >
          27 de mar. de 2023
        </time>
        <span
          class="separator"
        >
           | 
        </span>
        <span
          class="categories"
        >
          <span>
            <a
              href="/category/react"
            >
              React
            </a>
          </span>
          <span>
            <a
              href="/category/next"
            >
              Next
            </a>
          </span>
        </span>
      </p>
    </div>
  </header>
</div>
`;
